### PR TITLE
Refactor to use database as DNS ground truth, remove DNS worker code

### LINKED ISSUE #440: Replace DNS Record status UI/logic
The work in #396 is going to make it hard for us to continue with our DNS Record status UI and logic.  Instead, we're going to move to a world where we write to the database and then "reconcile" what is on Route53 in bulk--we won't update the status of individual records, since the database will be our source of truth.  As a result, we should fix our UI so that users can still get some info on whether or not their DNS record is fully propagated.

There are a few ways we could do this:

1. Write a Resource Route in Remix that allows us to send a list of domains, and it tries to resolve them server-side, and reports back with whether or not it worked for each one.  This is nice because we control it; however, it's also only reporting back what our server can reach vs. what's happening globally (DNS is distributed around the globe!)
2. Add a button that links users to an external site, and let it do this for us.  For example, I made a domain on staging, and you can test it at https://dnschecker.org/#A/foo.davidhumphrey.stage.mystudentproject.ca or https://www.nslookup.io/domains/foo.davidhumphrey.stage.mystudentproject.ca/dns-propagation/a/.  These sites (there are others, I just picked 2) let you construct a URL and have it do the queries for you using lots of DNS servers around the world.  This is nice because it gives a better picture of what's happening across many DNS servers.

### PR DESCRIPTION
**Pull request by humphd**
Fixes #440 

This switches over to use the database as our source of truth for all DNS Record info.  The reconciler is now responsible for making this happen in Route53, and we don't use DNS workers or direct calls anymore.

I've also fixed the bug where we didn't update the optional parts of a DNS Record in the db while editing.

I have to fix something with e2e tests, but I wanted to get this in fast, since it's blocking testing on Staging.

### CODE REVIEW

**Comment by humphd** [comment]
There might be more code we can remove that I'm missing.  Let me know if you see or think of anything.

File: app/components/dns-records-table.tsx [hunk 1]
```diff
@@ -121,7 +80,6 @@ export default function DnsRecordsTable(props: DnsRecordsTableProps) {
           <Table variant="striped" colorScheme="gray">
             <Thead>
               <Tr>
-                <Th />
                 <Th>Subdomain</Th>
```
Comment (Genne23v) [inline app/components/dns-records-table.tsx:83]: I think we should change this column name to DNS Record

**Review by Myrfion — APPROVED** [review]
Good to see have simpler things got and that we should care about the record status anymore!

**Review by Genne23v — CHANGES_REQUESTED** [review]

File: app/components/dns-records-table.tsx (hunk 1 above)
Comment (humphd) [reply app/components/dns-records-table.tsx:83]: Good idea, I'll do that.

**Comment by humphd** [comment]
Updated to not use `Subdomain`

**Review by Genne23v — APPROVED** [review]

**Review by Myrfion — APPROVED** [review]

**Review by dadolhay — CHANGES_REQUESTED** [review]
@humphd

When we are changing a record, I think we should call `setReconciliationNeeded()` from the `systemState` model

**Comment by humphd** [comment]
> @humphd
> 
> When we are changing a record, I think we should call `setReconciliationNeeded(true)` from the `systemState` model I think this could be integrated into the DNS model

Ah, indeed.  Changing now...

**Comment by humphd** [comment]
Updated to indicate that reconciler is needed after CRUD operations.

**Comment by cychu42** [comment]
> Updated to indicate that reconciler is needed after CRUD operations.

Do we need it for `renewDnsRecordById()` in `app/models/dns-record.server.ts`?

**Review by dadolhay — DISMISSED** [review]

**Comment by dadolhay** [comment]
> > Updated to indicate that reconciler is needed after CRUD operations.
> 
> Do we need it for `renewDnsRecordById()` in `app/models/dns-record.server.ts`?

No, that is only for our own system, telling when to delete the record

Us deleting the record from db, will trigger the reconciler to delete it from Route53

File: app/models/dns-record.server.ts [hunk 2]
```diff
@@ -48,7 +49,11 @@ export async function createDnsRecord(
   // Set expiration date 6 months from now
   const expiresAt = dayjs().add(6, 'month').toDate();
 
-  return prisma.dnsRecord.create({ data: { ...data, expiresAt } });
+  return prisma.dnsRecord.create({ data: { ...data, expiresAt } }).then((result) => {
```
[the remark this answers is not in the document]
Comment (humphd) [reply app/models/dns-record.server.ts:52]: Sure, I can.

**Review by Eakam1007 — DISMISSED** [review]

**Review by Myrfion — APPROVED** [review]

File: app/models/dns-record.server.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (humphd) [reply app/models/dns-record.server.ts:52]: Actually, I started to write it, and it's literally the same code.  I'm not sure adding a new function does much for us here.

**Comment by humphd** [comment]
Rebased and squashed.

**Review by Genne23v — APPROVED** [review]

**Review by Eakam1007 — APPROVED** [review]

**Pull request merged**